### PR TITLE
AArch64: Use stp/ldp instructions for preserved registers

### PR DESCRIPTION
**Pull request by knn-k**
This commit changes code generation for method prologue and epilogue to use the register pair instructions (stp and ldp) for saving and restoring preserved registers when applicable.

### CODE REVIEW

**Comment by knn-k** [comment]
Jenkins test sanity.functional,extended.functional alinux64,amac jdk17

**Comment by knn-k** [comment]
This commit changes the instructions in the method prologue like this.  It does the similar in the method epilogue.

```
  strimmx w24, [w20, 0]
  strimmx w25, [w20, 8]
  strimmx w26, [w20, 16]
  strimmx w27, [w20, 24]
  strimmx w28, [w20, 32]
    ↓
  stpoffx w24, w25, [w20, 0]
  stpoffx w26, w27, [w20, 16]
  strimmx w28, [w20, 32]
```

**Comment by knn-k** [comment]
Jenkins test sanity.functional,extended.functional alinux64,amac jdk17

**Comment by knn-k** [comment]
The failure in cmdLineTest_J9test_common_0 on AArch64 Linux looks like #13577.
https://openj9-jenkins.osuosl.org/job/Test_openjdk17_j9_sanity.functional_aarch64_linux_Personal/221/

```
Testing: -Xlockword minimizeFootprint mode
Test start time: 2023/04/17 04:05:50 Coordinated Universal Time
Running command: "/home/jenkins/workspace/Grinder/openjdkbinary/j2sdk-image/bin/java"   -Xdump -cp "/home/jenkins/workspace/Grinder/aqa-tests/TKG/../../jvmtest/functional/cmdLineTests/utils/utils.jar" -Xlockword:mode=minimizeFootprint VMBench.FibBench
Time spent starting: 4 milliseconds
Time spent executing: 2276 milliseconds
Test result: FAILED
Output from test:
 [OUT] Fibonacci: iterations = 10000
 [OUT] fibonacci(12) = 144
 [ERR] Unhandled exception
 [ERR] Type=Segmentation error vmState=0x00000000
 [ERR] J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000001
 [ERR] Handler1=0000FFFFAB5F2484 Handler2=0000FFFFAB559AC0 InaccessibleAddress=0000000000000024
```

File: runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp [hunk 1]
```diff
@@ -783,17 +783,45 @@ void J9::ARM64::PrivateLinkage::createPrologue(TR::Instruction *cursor)
 
       int32_t preservedRegisterOffsetFromJavaSP = outgoingArgsSize;
 
+      TR::RealRegister *regToStore = NULL;
```
Comment (0xdaryl) [inline runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp:786]: Minor nit: can you rename this to `pendingRegToStore` for improved readability?

**Review by 0xdaryl — COMMENTED** [review]
Looks good.  A couple of minor variable renames requested.

File: runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp (hunk 1 above)
Comment (knn-k) [reply runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp]: Changed as suggested.

File: runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp [hunk 2]
```diff
@@ -881,16 +909,42 @@ void J9::ARM64::PrivateLinkage::createEpilogue(TR::Instruction *cursor)
    int32_t preservedRegisterOffsetFromJavaSP = cg()->getLargestOutgoingArgSize() + getOffsetToFirstParm(); // outgoingArgsSize
    TR::RealRegister::RegNum firstPreservedGPR = TR::RealRegister::x21;
    TR::RealRegister::RegNum lastPreservedGPR = TR::RealRegister::x28;
+   TR::RealRegister *regToLoad = NULL;
```
[the remark this answers is not in the document]
Comment (knn-k) [reply runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp]: Changed as suggested.

**Comment by knn-k** [comment]
Jenkins test sanity.functional alinux64,amac jdk17

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**